### PR TITLE
WEBDEV-6174 Add models for collection page extra info

### PR DESCRIPTION
**Pull request by latonv**
The PPS now provides `collection_extra_info` for `collection_details` requests, and will soon allow multiple hit types to be included in a single response. This PR updates the search service models to accommodate the new response changes.

### CODE REVIEW

File: src/models/hit-types/hit.ts [hunk 1]
```diff
@@ -8,16 +8,19 @@ import type { TextHit } from './text-hit';
...
- * Really just to keep things clean at the call site.
+ * Additional information provided by the PPS about hits, separately from
+ * their fields map.
  */
-interface PreserveAlias {} // eslint-disable-line @typescript-eslint/no-empty-interface
+interface HitInfo {
+  hit_type?: HitType;
+  index?: string;
+  service_backend?: string;
+}
```
Comment (latonv) [inline src/models/hit-types/hit.ts:18]: An imminent PPS upgrade (allowing multiple hit types in one response) will add these properties to hits. Bonus: we no longer need this little type-hack from before.

File: test/responses/search-response-details.test.ts [hunk 2]
```diff
@@ -131,4 +177,35 @@ describe('SearchResponseDetails', () => {
...
+      field_properties: {},
+    };
+
+    const details = new SearchResponseDetails(
+      responseBodyWithoutExtraInfo,
+      responseSchema
+    );
+    expect(details.results.length).to.equal(2);
+    expect(details.collectionExtraInfo).to.be.undefined;
+  });
```
Comment (iisa) [inline test/responses/search-response-details.test.ts:210]: 👏

**Review by iisa — APPROVED** [review]
looks good!

File: test/responses/search-response-details.test.ts [hunk 3]
```diff
@@ -30,6 +30,9 @@ const responseBody: SearchResponseBody = {
   collection_titles: {
     baz: 'Baz Collection',
   },
+  collection_extra_info: {
+    thumbnail_url: 'foo',
```
[the remark this answers is not in the document]
Comment (latonv) [reply test/responses/search-response-details.test.ts:34]: Thanks for noting this -- we can bring it up to discuss w/ the team. Atm the image service doesn't always give us the right image, while the PPS-provided `thumbnail_url` should always be what we need.

**Pull request merged**